src/Nethermind/Nethermind.Pipeline/IPipepline.cs

```diff
@@ -0,0 +1,14 @@
+using Nethermind.Pipeline.Publishers;
+using Nethermind.PubSub;
+using System.Collections.Generic;
+
+namespace Nethermind.Pipeline
+{
+    public interface IPipeline<T>
+    {
+        ISource<T> Source { get; }
+        Stack<IPipelineElement<T>> Elements { get; }
```

Comment:
better LinkedList

Reply:
look at the DotNetty pipelines implementations -> may serve as an example
PipelineElement for sure should be <TIn, TOut>